Mathlib/LinearAlgebra/Matrix/GeneralLinearGroup.lean

```diff
@@ -146,6 +146,32 @@ theorem toLinear_apply (v : n → R) : (toLinear A).toLinearEquiv v = Matrix.mul
 
 end CoeLemmas
 
+variable {S T : Type*} [CommRing S] [CommRing T]
+
+/-- A ring homomorphism ``f : R →+* S`` induces a homomorphism ``GLₙ(f) : GLₙ(R) →* GLₙ(S)``. -/
+def map  (f : R →+* S) : GL n R →* GL n S where
+  toFun g :=
+    { val := f.mapMatrix g
+      inv := f.mapMatrix (Units.inv g)
+      val_inv := by rw [← RingHom.map_mul, Units.val_inv, RingHom.map_one _]
+      inv_val := by rw [← RingHom.map_mul, Units.inv_val, RingHom.map_one _] }
+  map_one' := by aesop
+  map_mul' x y := by aesop
+
+@[simp]
+theorem map_id :
+    map (RingHom.id R) = MonoidHom.id (GL n R) := by
+  rfl
+
+@[simp]
+theorem map_comp (f : T →+* R) (g : R →+* S) :
+    map (g.comp f) = (map g).comp (map (n := n) f) :=
+  rfl
+
+@[simp]
+theorem map_comp_apply (f : T →+* R) (g : R →+* S) (x : GL n T) :
+    (map g).comp (map f) x = map g (map f x) := rfl
```

Comment:
These last two proofs: why do you have one `rfl` on the same line and one on the next line? I don't know what the style guide says but I suspect it would like consistency :-) Maybe look up what the recommended style is?